### PR TITLE
Add support for instanceof

### PR DESCRIPTION
**Pull request by j-f1**
~This depends on #24. I’ll rebase when that’s done.~

### CODE REVIEW

**Comment by j-f1** [comment]
Update: I was able to separate this completely from #24

File: Runtime/src/index.ts [hunk 1]
```diff
@@ -362,6 +362,15 @@ export class SwiftRuntime {
...
                 writeUint32(result_obj, this.heap.allocHeap(result));
             },
+            swjs_instanceof: (
+                obj_ref: ref, constructor_ref: ref,
+                result_ptr: pointer
+            ) => {
+              const obj = this.heap.referenceHeap(obj_ref)
+              const constructor = this.heap.referenceHeap(constructor_ref)
+              const result = obj instanceof constructor
+              writeUint32(result_ptr, Number(result))
```
Comment (kateinoigakukun) [inline Runtime/src/index.ts:372]: Why don't you just return the result value?
```suggestion
              return obj instanceof constructor
```

File: Sources/JavaScriptKit/JSObject.swift [hunk 2]
```diff
@@ -32,6 +32,16 @@ public class JSObjectRef: Equatable {
         getJSValue(this: self, index: Int32(index))
     }
 
+    public func instanceof(_ constructor: JSValue) -> Bool {
+        instanceof(constructor.function!)
```
Comment (kateinoigakukun) [inline Sources/JavaScriptKit/JSObject.swift:36]: This library shouldn't provide unsafe method

File: Runtime/src/index.ts (hunk 1 above)
Comment (j-f1) [reply Runtime/src/index.ts]: Oh, cool! Wasn’t sure if this was possible

File: Sources/JavaScriptKit/JSObject.swift (hunk 2 above)
Comment (j-f1) [reply Sources/JavaScriptKit/JSObject.swift]: I changed it to default to `false` if you don’t pass a function in. Does that make sense?

Comment (kateinoigakukun) [reply Sources/JavaScriptKit/JSObject.swift]: I think this convenience method is unnecessary. The users should handle the non-function case.

Comment (MaxDesiatov) [reply Sources/JavaScriptKit/JSObject.swift]: I agree, `new` is similarly defined on `JSFunctionRef`, not `JSObjectRef` IIRC

**Pull request merged**